deploy/helm/elastic-agent/templates/agent/_outputs.tpl

```diff
@@ -127,21 +127,141 @@ namespace: {{.}}
 {{/* no preset env vars for ESECKRef output */}}
 {{- end -}}
 
+{{- define "elasticagent.output.Logstash.preset.config" -}}
+{{- $ := index . 0 -}}
+{{- $outputName := index . 1 -}}
+{{- $outputVal := deepCopy (index . 2) -}}
+{{$outputName}}:
+  hosts:
+  {{- range $idx, $host := $outputVal.hosts }}
+    - {{$host | quote}}
+  {{- end }}
+  type: logstash
+  {{- if hasKey $outputVal "loadbalance" }}
+  loadbalance: {{$outputVal.loadbalance}}
+  {{- end }}
+  {{- with $outputVal.ttl }}
+  ttl: {{. | quote}}
+  {{- end }}
+  {{- if hasKey $outputVal "slow_start" }}
+  slow_start: {{ $outputVal.slow_start}}
+  {{- end }}
+  {{- with $outputVal.pipelining }}
+  pipelining: {{.}}
+  {{- end }}
+  {{- with $outputVal.workers }}
+  workers: {{.}}
+  {{- end }}
+  {{- with $outputVal.timeout }}
+  timeout: {{. | quote }}
+  {{- end }}
+
+  {{- with index $outputVal "queue.mem.flush.timeout" }}
+  queue.mem.flush.timeout: {{. | quote}}
+  {{- end }}
+  {{- with index $outputVal "queue.mem.flush.min_events" }}
+  queue.mem.flush.min_events: {{.}}
+  {{- end }}
+  {{- with index $outputVal "queue.mem.events" }}
+  queue.mem.events: {{.}}
+  {{- end }}
```

Comment:
I believe that there shouldn't be any fields with dot notation in them here. If a user sets through cli such values they will appear as proper yaml and then this here won't work

Reply:
I think doing the deprecation may be best in a separate PR otherwise this one will get event longer. I have made the other changes suggested.